msvc/task/GenVersionFiles/RCHeaderFile.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.IO;
+
+namespace GenVersionFiles
+{
+    class RCHeaderFile
+    {
+        public RCHeaderFile(VersionInfo version)
+        {
+            this.version = version;
+        }
+
+        private VersionInfo version;
+
+        public void Write(string filePath)
+        {
+            using (StreamWriter file = new System.IO.StreamWriter(filePath))
+            {
+                file.WriteLine("#define FILE_VERSION {0}, {1}, {2}, {3}", version.Major, version.Minor, version.Build, version.Revision);
```

Comment:
SharedVersion.h:
#define FILE_VERSION 0, 2, 1610, 24
#define FILE_VERSION_STR "0.2.1610.24"

SharedAssemblyInfo.cs:
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

[assembly: AssemblyFileVersion("0.2.1610.24")]
[assembly: AssemblyInformationalVersion("0.2.1610.24")]

---
In reply to: [84728270](https://github.com/Microsoft/WinObjC/pull/1184#discussion_r84728270) [](ancestors = 84728270)